samples/javascript_nodejs/51.teams-messaging-extensions-action/bots/teamsMessagingExtensionsActionBot.js

```diff
@@ -1,45 +1,139 @@
 // Licensed under the MIT License.
 
-const { TeamsActivityHandler, CardFactory } = require('botbuilder');
+const { TeamsActivityHandler, CardFactory, TeamsInfo } = require('botbuilder');
 
 class TeamsMessagingExtensionsActionBot extends TeamsActivityHandler {
     handleTeamsMessagingExtensionSubmitAction(context, action) {
         switch (action.commandId) {
-        case 'createCard':
-            return createCardCommand(context, action);
-        case 'shareMessage':
-            return shareMessageCommand(context, action);
-        default:
-            throw new Error('NotImplemented');
+            case 'createCard':
+                return createCardCommand(context, action);
+            case 'shareMessage':
+                return shareMessageCommand(context, action);
+            default:
+                throw new Error('NotImplemented');
         }
     }
+
+    async handleTeamsMessagingExtensionFetchTask(context, action) {
+        var member;
+        try {
+            member = await this.getSingleMember(context);
+        } catch (e) {
+            if (e.code === 'BotNotInConversationRoster') {
+                return {
+                    task: {
+                        type: 'continue',
+                        value: {
+                            card: GetJustInTimeCardAttachment(),
+                            height: 400,
+                            title: 'Adaptive Card - App Installation',
+                            width: 300
+                        },
+                    },
+                };
+            }
+            throw e;
+        }
+        return {
+            task: {
+                type: 'continue',
+                value: {
+                    card: GetAdaptiveCardAttachment(),
+                    height: 400,
+                    title: 'Hello ' + member,
+                    width: 300
+                },
+            },
+        };
+    }
+    async getSingleMember(context) {
+        var member;
+        try {
+            member = await TeamsInfo.getMember(
+                context,
+                context.activity.from.id
+            );
+        } catch (e) {
+            if (e.code === 'MemberNotFoundInConversation') {
+                context.sendActivity(MessageFactory.text('Member not found.'));
+                return e.code;
+            } else {
+                console.log(e);
+                throw e;
+            }
+        }
+        return member.name;
+    }
+}
+
+function GetJustInTimeCardAttachment() {
+    return CardFactory.adaptiveCard({
+        actions: [
+            {
+                type: 'Action.Submit',
+                title: 'Continue',
+                data: { msteams: { justInTimeInstall: true } }
+            },
+        ],
+        body: [
+            {
+                text:
+                    'Looks like you have not used Action Messaging Extension app in this team/chat. Please click **Continue** to add this app.',
```

Comment:
```suggestion
                text: 'Looks like you have not used Action Messaging Extension app in this team/chat. Please click **Continue** to add this app.',
```

You probably got a linter warning here, but IMO, it would be better to just have a single, slightly longer line.

Reply:
Changed this to one line